### DISCUSSION TITLE
Python interfaces to C++ (and others)

### DISCUSSION THREAD

**Discussion by davidorme** [ImperialCollegeLondon/virtual_rainforest#2] [Q&A] [answered]
## Background

We are definitely going to want to interface with C++ code right from the start. 

@jaideep777 is developing a plant growth model (https://jaideep777.github.io/Plant-FATE/) and a ODE solver for modelling plant demographic processes (https://jaideep777.github.io/libpspm/), both of which are are written in C++.

Given that there are quite a lot of frameworks for creating Python / C++ interfaces:

## Questions

**Does anyone have any recommendations on the best Python / C++ interface to use?**

**Comment by cc-a** [comment]
I'm not in a position to say which is best but the one we have hands-on experience with using is [Swig](https://www.swig.org/).

**Comment by dalonsoa** [comment, marked as answer]
We have a new member of our team joining us at the end of June and that I was planing to put to work in this project because of his extensive experience doing precisely that. I'd suggest to postpone a decision about the best approach until he's here.

**Reply by jaideep777** [reply]
Actually, Plant-FATE now has a python interface based on pybind11.

**Comment by dalonsoa** [comment]
I think his expertise was [pybind11](https://github.com/pybind/pybind11), actually.
